scripts/romanisim-make-stack

```diff
@@ -0,0 +1,267 @@
+
+                    if args.debug:
+                        print(f"output_file_name = {output_file_name}")
+
+                    # If making a script, write the line and proceed
+                    # otherwise create simulation file
+                    if args.make_script:
+                        line = f"romanisim-make-image {output_file_name}"
+
+                        for option, value in options_dct.items():
+                            if (type(value) != bool) and (option != 'sca'):
+                                line += f" --{option} {value}"
+                            elif (option == 'sca'):
+                                line += f" --{option} {sca}"
+                            elif (value == True):
+                                line += f" --{option}"
+
+                        line += f" --bandpass {entry['optical_element']}"
+                        line += f" --radec {entry['ra']} {entry['dec']}"
+                        line += f" --ma_table_number {entry['ma_table_number']}"
+                        line += f" --date "
+                        for s in date.strftime('%Y %m %d %H %M %S %f').split(' '):
+                            line += f"{int(s)} "
+
+                        # print line
+                        if args.debug:
+                            print(f'line = {line}')
+
+                        # add line to script
+                        spcript_file.write(line+'\n')
+
+                    else:
+                        # Obtain WCS catalog
+                        if args.catalog is None:
+                            twcs = wcs.get_wcs(metadata, usecrds=args.usecrds)
+                            rd_sca = twcs.toWorld(galsim.PositionD(
+                                galsim.roman.n_pix / 2, galsim.roman.n_pix / 2))
+                            cat = catalog.make_dummy_table_catalog(
+                                rd_sca, bandpasses=[entry['optical_element']], nobj=args.nobj)
+                        else:
+                            log.warning('Catalog input will probably not work unless the catalog '
+                                        'covers a lot of area or you have thought carefully about '
+                                        'the relation between the boresight and the SCA locations.')
+                            cat = Table.read(args.catalog)
+
+                        # If selected, apply persistence
+                        if args.previous:
+                            # Use provided Persistence file for first SCA (for each entry) only
+                            # after that, use previously created file
+                            if args.sca > 0 and sca > 1:
+                                persist = persistence.Persistence.read(previous_file_name)
+                            else:
+                                persist = persistence.Persistence.read(args.previous)
```

Comment:
Persistence is specific to each detector.  We need to pass in here the previous image simulated in the appropriate SCA.  The easiest approach is probably to make previous_file_name a dict, to fill in previous_file_name[sca] = output_file_name, and to do
prevfn = previous_file_name.get(sca, None)
persist = persist.Persistence() if prevfn is None else Persistence.read(prevfn)

Reply:
Agreed, I shall make it so.

Reply:
Implemented.